internal/config/auth.go

```diff
@@ -0,0 +1,41 @@
+package config
+
+// Auth represents authentification options.
+// TODO Auth options should generate valid header for request
+type Auth interface {
+	validate() error
+}
+
+type errInvalidAuth struct{}
```

Comment:
This could be just a variable (if we don't use custom messages). Also, it could be useful to declare errors as public to check it later with `errors.As()` or `errors.Is()`:
```suggestion
var ErrInvalidAuth = errors.New("Invalid authentication")
```

Later, to customize error message, you can use `%w` formatting symbol: `fmt.Errorf("No token provided: %w", err)`, this will pass `errors.As/Is` checks too:
```go
err := fmt.Errorf("Invalid password: %w", ErrInvalidAuth)
errors.Is(err, ErrInvalidAuth) // true
```